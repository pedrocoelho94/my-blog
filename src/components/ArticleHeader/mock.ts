import { data } from 'api/mockPosts.json'

const { id, title, excerpt, cover, author, categories, createdAt } =
  data.posts[0]

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}
