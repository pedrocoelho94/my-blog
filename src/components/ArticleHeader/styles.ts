import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
