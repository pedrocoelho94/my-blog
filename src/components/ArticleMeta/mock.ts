export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'pedro-coelho',
    displayName: 'Pedro Coelho'
  },
  categories: [
    {
      id: '78724ec23f',
      displayName: 'Tech',
      slug: 'tech'
    },
    {
      id: '762vg214g34',
      displayName: 'JS',
      slug: 'javascript'
    }
  ]
}
