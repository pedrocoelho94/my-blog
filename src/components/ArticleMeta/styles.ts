import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
