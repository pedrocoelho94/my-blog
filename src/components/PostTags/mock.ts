import { data } from 'api/mockPosts.json'

export default { tags: data.posts[0] }.tags
