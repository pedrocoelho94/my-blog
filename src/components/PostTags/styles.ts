import styled, { css } from 'styled-components'

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin-right: 0.5rem;
    }

    span:not(:last-child)::after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
