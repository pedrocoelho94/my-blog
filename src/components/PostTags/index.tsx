import Link from 'next/link'

import { PostTag } from 'shared-typed/tag'
import * as S from './styles'

export type PostTagsProps = {
  tags?: PostTag[]
}

const PostTags = ({ tags = [] }: PostTagsProps) => {
  if (tags.length === 0) return null

  // organiza as tags em ordem alfabetica
  tags.sort((a, b) =>
    a.displayName > b.displayName ? 1 : b.displayName > a.displayName ? -1 : 0
  )

  return (
    <S.Wrapper>
      Saiba mais:{' '}
      {tags?.map((tag) => (
        <span key={tag.id}>
          <Link href={`/tag/${tag.slug}`} passHref>
            <a>{tag.displayName}</a>
          </Link>
        </span>
      ))}
    </S.Wrapper>
  )
}

export default PostTags
