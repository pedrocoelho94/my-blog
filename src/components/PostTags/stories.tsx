import { Story, Meta } from '@storybook/react'
import PostTags, { PostTagsProps } from '.'

import mock from './mock'

export default {
  title: 'PostTags',
  component: PostTags,
  args: mock
} as Meta

export const Default: Story<PostTagsProps> = (args) => (
  <div>
    <PostTags {...args} />
  </div>
)

export const WithoutTags: Story<PostTagsProps> = (args) => (
  <div>
    <PostTags {...args} tags={[]} />
  </div>
)
