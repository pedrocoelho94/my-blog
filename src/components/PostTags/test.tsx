import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'

import PostTags, { PostTagsProps } from '.'
import mock from './mock'

const props: PostTagsProps = mock

describe('<PostTags />', () => {
  it('should render two tags', () => {
    const { container } = renderWithTheme(<PostTags {...props} />)

    expect(screen.getAllByRole('link')).toHaveLength(2)

    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with no tags', () => {
    const { container } = renderWithTheme(<PostTags {...props} tags={[]} />)

    expect(container).toMatchSnapshot()
  })
})
