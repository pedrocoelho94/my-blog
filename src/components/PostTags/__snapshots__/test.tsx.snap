// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should match snapshot with no tags 1`] = `<div />`

exports[`<PostTags /> should render two tags 1`] = `
.c0 {
  margin: 2.4rem 0;
}

.c0 span {
  margin-right: 0.5rem;
}

.c0 span:not(:last-child)::after {
  content: ', ';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(0.5);
  filter: brightness(0.5);
}

<div>
  <p
    class="c0"
  >
    Saiba mais:
     
    <span>
      <a
        href="/tag/barcelona"
      >
        Barcelona
      </a>
    </span>
    <span>
      <a
        href="/tag/xavi"
      >
        Xavi
      </a>
    </span>
  </p>
</div>
`
